Documentation/k8s-object.md

```diff
@@ -80,53 +80,63 @@ See the [Object Storage](client.md#object-storage) documentation for more steps
 ## Access External to the Cluster
 
 Rook sets up the object storage so pods will have access internal to the cluster. If your applications are running outside the cluster,
-you will need to setup an external service through a `NodePort`.
+you will need to setup an ingress controller.
 
-First, note the service that exposes RGW internal to the cluster. We will leave this service intact and create a new service for external access.
+First, note the service that exposes RGW internal to the cluster. We will leave this service intact.
 ```bash
 $ kubectl -n rook get service rook-ceph-rgw-my-store
 NAME                     CLUSTER-IP   EXTERNAL-IP   PORT(S)     AGE
 rook-ceph-rgw-my-store   10.3.0.177   <none>        80/TCP      2m
 ```
 
-Save the external service as `rgw-external.yaml`:
+Save the ingress resource as `rgw-external.yaml`:
 
 ```yaml
-apiVersion: v1
-kind: Service
+apiVersion: extensions/v1beta1
+kind: Ingress
 metadata:
-  name: rook-ceph-rgw-my-store-external
+  name: rgw-my-store-external
   namespace: rook
-  labels:
-    app: rook-ceph-rgw
-    rook_cluster: rook
-    rook_object_store: my-store
+  annotations:
+    kubernetes.io/ingress.class: "nginx"
 spec:
-  ports:
-  - name: rgw
-    port: 53390
-    protocol: TCP
-    targetPort: 53390
-  selector:
-    app: rook-ceph-rgw
-    rook_cluster: rook
-    rook_object_store: my-store
-  sessionAffinity: None
-  type: NodePort
+  rules:
+  - host: my-host
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: rook-ceph-rgw-my-store
+          servicePort: 80
 ```
 
-Now create the external service.
+Create the ingress resource.
 
 ```bash
 kubectl create -f rgw-external.yaml
 ```
 
-See both rgw services running and notice what port the external service is running on:
+See the ingress controller:
```

Comment:
This makes it seem that using ingress is the only way to consume the object store. I think it is up to the users to decide how the object store service is consumed. It could be done with ingress, nodeport, cloud loadbalancer etc

Reply:
NodePort to me in Kubernetes has always felt like a hack, one would never tell someone in real life "hey, go to 192.168.156.5:34590" for the object store, people should be going: "The object store is at http://rgw.k8s-platform.company.com".

Reply:
this is true. But if the client application is from another pod in kubernetes, doing nodeport or ingress is not necessary. I guess what im trying to get at is it is up to the users to decide if:
1- the ports should be accessible externally. If not, then that is not needed and the simple Service resource should suffice.
2- If external access is needed then nodeport, loadbalancer or ingress would work

I just want to avoid giving the impression that ingress is the only way for getting ObjectStore running.